opentech/apply/categories/blocks.py

```diff
@@ -58,7 +59,7 @@ def get_field_kwargs(self, struct_value):
         kwargs = super().get_field_kwargs(struct_value)
         category = struct_value['category']
         kwargs = self.use_defaults_from_category(kwargs, category)
-        choices = category.options.values_list('id', 'value')
+        choices = category.options.filter(Q(is_archived=False) | Q(id__in=[])).values_list('id', 'value')
```

Comment:
@Parbhat Possible to get the current values in the form at this stage? Want to filter on "id__in" so to show only archived terms if they are set on the submission.

Reply:
Not exactly here. This piece of code only runs while generating form fields for the form. It looks okay for new form submission but on submission edit page it will not display archived answers. The existing archived options values should display correctly in submission data until the user hit save on edit page. On the edit page, we can also add the archived choices if is in submitted data by updating [initial](https://github.com/OpenTechFund/opentech.fund/blob/c9fb73f3d79391ad2dcdb4ce780ccda336bce588/opentech/apply/funds/views.py#L654) values in BaseSubmissionEditView.

Reply:
Thanks, now I know what I need to do.